PyInstaller/isolated/_parent.py

```diff
@@ -110,17 +195,34 @@ def __init__(self):
         self._child = None
 
     def __enter__(self):
-        # We need two pipes. One for the child to send data to the parent.
-        self._read_from_child, self._write_to_parent = pipe()
-        # And one for the parent to send data to the child.
-        self._read_from_parent, self._write_to_child = pipe()
+        # We need two pipes. One for the child to send data to the parent. The (write) end-point passed to the
+        # child needs to be marked as inheritable.
+        read_from_child, write_to_parent = create_pipe(False, True)
+        # And one for the parent to send data to the child. The (read) end-point passed to the child needs to be
+        # marked as inheritable.
+        read_from_parent, write_to_child = create_pipe(True, False)
 
         # Spawn a Python subprocess sending it the two file descriptors it needs to talk back to this parent process.
-        self._child = child(self._read_from_parent, self._write_to_parent)
-
-        # Open file handles to talk to the child.
-        self._write_handle = open(self._write_to_child, "wb")
-        self._read_handle = open(self._read_from_child, "rb")
+        self._child = child(read_from_parent, write_to_parent)
+
+        # Close the end-points that were inherited by the child.
+        close_pipe_endpoint(read_from_parent)
+        close_pipe_endpoint(write_to_parent)
+        del read_from_parent
+        del write_to_parent
```

Comment:
You're just deleting these to stop them accidentally being used, right? There's no ugly pipe closing kink which requires the handles to be garbage collected before the pipe can be truly closed.